mlflow/store/s3_artifact_repo.py

```diff
@@ -69,15 +69,29 @@ def list_artifacts(self, path=None):
         for result in results:
             # Subdirectories will be listed as "common prefixes" due to the way we made the request
             for obj in result.get("CommonPrefixes", []):
-                subdir = obj.get("Prefix")[len(artifact_path)+1:]
-                if subdir.endswith("/"):
-                    subdir = subdir[:-1]
-                infos.append(FileInfo(subdir, True, None))
+                subdir_path = obj.get("Prefix")
+                if not subdir_path.startswith(artifact_path):
+                    raise ValueError(
```

Comment:
**nit** 
Why raise a `ValueError`? can we raise `MlflowException`

Reply:
Changed to `MlflowException`!